frameworklauncher/src/main/java/com/microsoft/frameworklauncher/applicationmaster/StatusManager.java

```diff
@@ -493,6 +495,30 @@ private void logTaskStateCounters(String taskRoleName) {
   /**
    * REGION ReadInterface
    */
+
+  public Integer getUnAllocatedTaskCount(String taskRoleName) {
+    int unAllocatedTastCount = 0;
+    List<TaskStatus> taskStatusList = taskStatuseses.get(taskRoleName).getTaskStatusArray();
+    for (TaskStatus taskstatus : taskStatusList) {
+      if (taskstatus.getTaskState() == TaskState.TASK_WAITING || taskstatus.getTaskState() == TaskState.CONTAINER_REQUESTED) {
+        unAllocatedTastCount++;
```

Comment:
break to 2 lines

Reply:
Use CONTAINER_ASSOCIATED_STATES to filter

---
In reply to: [173099566](https://github.com/Microsoft/pai/pull/318#discussion_r173099566) [](ancestors = 173099566)